app/src/main/java/org/signal/cashu/service/DefaultCashuService.kt

```
@@ -0,0 +1,145 @@
+                .post(requestBody)
+                .build()
+
+            val response = httpClient.newCall(request).execute()
+            if (response.isSuccessful) {
+                transactionDao.insertTransaction(transaction.copy(status = TransactionStatus.COMPLETED))
+                return true
+            } else {
+                transactionDao.insertTransaction(transaction.copy(status = TransactionStatus.FAILED))
+                return false
+            }
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)